### ISSUE TITLE
Allow module extensions to fail on a tag

### ISSUE THREAD

**Issue by fmeum** [bazelbuild/bazel#17375]
### Description of the feature request:

Module extensions should be able to fail on a particular module extension tag so that the emitted stack trace points to the location at which the particular tag was created by the user.

### What underlying problem are you trying to solve with this feature?

When a user specifies an invalid module extension tag in their `MODULE.bazel` file (e.g. a version of a dependency that isn't valid) and the module extensions `fail`s due to that, the stack trace starts with the module extension implementation function. There is no way to reference the particular tag that caused the failure, which makes it unnecessarily difficult for the user to fix the issue.

### Which operating system are you running Bazel on?

Any

### What is the output of `bazel info release`?

6.0.0

**Comment by meteorcloudy** [comment]
@fmeum Is this already fixed?

**Comment by fmeum** [comment]
@meteorcloudy Yes, it was fixed in https://github.com/bazelbuild/bazel/pull/19274.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 6.4.0 RC1](https://releases.bazel.build/6.4.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
